adapters/improvedigital/improvedigital.go

```diff
@@ -144,3 +153,68 @@ func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType,
 		Message: fmt.Sprintf("Failed to find impression for ID: \"%s\"", impID),
 	}
 }
+
+// This method responsible to clone request and convert additional consent providers string to array when additional consent provider found
+func (a *ImprovedigitalAdapter) getAdditionalConsentProvidersUserExt(request openrtb2.BidRequest) []byte {
+	const (
+		consentProvidersSettingsInputKey = "ConsentedProvidersSettings"
+		consentProvidersSettingsOutKey   = "consented_providers_settings"
+		consentedProvidersKey            = "consented_providers"
+	)
+
+	var (
+		err   error
+		cpStr string
+	)
+
+	// If user/user.ext not defined, no need to parse additional consent
+	if request.User == nil || request.User.Ext == nil {
+		return []byte{}
+	}
+
+	// Start validating additional consent
+	// Check key exist user.ext.ConsentedProvidersSettings
+	var userExtMap = make(map[string]json.RawMessage)
+	if err = json.Unmarshal(request.User.Ext, &userExtMap); err != nil {
+		return []byte{}
+	}
+
+	cpsMapValue, cpsJSONFound := userExtMap[consentProvidersSettingsInputKey]
+	if !cpsJSONFound {
+		return []byte{}
+	}
+
+	// Check key exist user.ext.ConsentedProvidersSettings.consented_providers
+	var cpMap = make(map[string]json.RawMessage)
+	if err = json.Unmarshal(cpsMapValue, &cpMap); err != nil {
+		return []byte{}
+	}
+
+	cpMapValue, cpJSONFound := cpMap[consentedProvidersKey]
+	if !cpJSONFound {
+		return []byte{}
```

Comment:
Is there a way to test this error case? If so could you update the tests to cover it.